### ISSUE TITLE
Question Regarding to Appendix H

### ISSUE THREAD

**Issue by EmreTaha** [rdangovs/essl#3]
Hi,
 Very nice work. I have a small question regarding to appendix H. In there you stated that It is problematic if x and T_g(x) exist in the data. Why is it problematic exactly in a sense that the labels are different due to the orientation and the change in the representational space due to the equivariance should help the downstream task.

**Comment by rdangovs** [comment]
Hello,

This is a very good question. Indeed, if x and T_g(x) exist in the data, then the labels of their orientation are different. Standard E-SSL will treat both x and T_g(x) with the same label, which is an issue. Relative orientation prediction lifts that issue, as we demonstrate with preliminary studies in Appendix H.

This was an interesting experiment, motivated by reviewer 9Qn8 in the following discussion (https://openreview.net/forum?id=gKLAAfiytI). 

You can think that ESSL, the main version, posed in the paper, would be most useful if G "breaks the symmetry" of the data, i.e. if T_g(x) is outside of the dataset. That would remove potential ambiguities with the classification objective that encourages sensitivity.

Happy to chat more about it :)

**Comment by EmreTaha** [comment]
Thanks a lot for the explanation. Especially after reading the discussion with the reviewer I think I understand a little bit better. Just to check If I understood correctly, the reviewer said:
>  To the authors credit, it is spelled out explicitly in Proposition 1, but I think it is down-played a bit, when it should probably be explored more. Taking the example of 90 degree rotations, the assumption is that every image has a correctly inferable orientation. This is, in fact, what the loss of eqn. (2) is learning. So if both a datapoint x and a datapoint rot(x) exist in the dataset, the model cannot learn eqn (2) correctly for both.

So my take from this is that, we should be able to deduce T'_g from [f(x), f(T_g(x))],  which the predictor learns the relative relation. Else the image specific representations would be weird since it would try to learn about T'_g(x) from both f(T_g(x)) and its counterpart in the dataset f(T_g(rot(x)). But if we rotate the image as any degree possible and the equivariance task is a regression task, and it rarely sees the same rot(x) and x, do you think it make sense?

Another follow-up question. So sensitivity (equivariant) transformations should form a group. Is that a condition for T'_g(x) to be non-parametric transformation? Because in one of the very early works ([Slow and steady feature analysis: higher order temporal coherence in video](https://arxiv.org/abs/1506.04714)), they assumed that if the T is defined in a relatively small neighborhood, T' can be assumed as non-parametric (page 5, first paragraph). it is a video classification task with temporal equivariance in very short clips. It can be said that transformations of the frame scenes through the time do not constitute a group. Then, to reiterate the question; in order to have a non-parametric T'_g(x) (temporal), the transformations do not need to be invertible, associative (i.e. not necessarily form a group). Else, aka the "group" is a condition, the formulation in that work is weak. 
It is a little bit long, but I hope I managed to convey what I tried to say x) 

Edit: I realized that this iccv'21 [paper](https://arxiv.org/abs/2112.03624) addresses the same issue (app. H) and solves similarly (sec. 3.1).

**Comment by rdangovs** [comment]
> So my take from this is that, we should be able to deduce T'_g from [f(x), f(T_g(x))], which the predictor learns the relative relation. Else the image specific representations would be weird since it would try to learn about T'_g(x) from both f(T_g(x)) and its counterpart in the dataset f(T_g(rot(x)). But if we rotate the image as any degree possible and the equivariance task is a regression task, and it rarely sees the same rot(x) and x, do you think it make sense?

Indeed, I belive your understanding aligns with mine. I think this issue is alleviated when the rotation is continuous and the task is regression, I agree with you. It might be interesting to try both E-SSL and the modification in Appendix H. I suspect that both versions might perform similarly.

> Is that a condition for T'_g(x) to be non-parametric transformation?

It doesn't have to be a group. We gave examples with groups in my paper and used the formalism of group actions, but you can apply the ESSL loss function to any types of action. You can even make T'_g parametric and learn it :).

> Edit: I realized that this iccv'21 [paper](https://arxiv.org/abs/2112.03624) addresses the same issue (app. H) and solves similarly (sec. 3.1).

This paper is interesting indeed. My understanding is that they concatenate the features from the original frame and its temporal transform, and then they do contrastive learning on the concatenation. However, we predict the transformation in Appendix H, and the contrastive learning is not done on the concatenation. Indeed, there are multiple approaches to the problem, it would be interesting to see a systematic study around that.

**Comment by EmreTaha** [comment]
It is more clear now. I realized that I needed to do more reading :)
I think the main problem with time related equivariance is that the transformation parametrized by the time is not accessible in the image domain (and it is not symmetric as well, like new information is introduced some removed, so you cannot model it with morph map etc.. maybe optical flow in videos), but could be learned or modeled (as trajectory or displacement map). It became more clearer after reading this paper actually https://arxiv.org/abs/2202.07559 :)
In the representation space, I think nowadays people moved to[ transformed representation prediction ](https://openreview.net/forum?id=eDLwjKmtYFt) and [generating the parameters of the transformation applied to the representation](https://arxiv.org/abs/2302.10283) from recovering the image transformation parameter. I think for the transformation related to time, predicting the next time-steps representation is more suitable.
Thanks a lot for your help and answers, it cleared a lot for me. I am looking forward to your future work.
Also if you have more suggestions (papers, books etc.) it would be most appreciated ;)
